stock_release_channel_partner_address/tests/test_release_channel_partner_address.py

```diff
@@ -0,0 +1,120 @@
+            {
+                "name": "Partner IT Ancona",
+                "country_id": cls.italy.id,
+                "state_id": cls.ancona.id,
+            }
+        )
+
+        # Prepare channels by country and state
+        # First, make sure all existing channels are archived (to avoid issues w/
+        # preexisting records)
+        cls.env["stock.release.channel"].search([]).action_archive()
+        # Then, create a channel for each possible combination of country and state
+        cls.channel = cls.env["stock.release.channel"].create(
+            {
+                "name": "Channel",
+                "country_ids": [],
+                "state_ids": [],
+                "state": "open",
+            }
+        )
+        cls.channel_it = cls.env["stock.release.channel"].create(
+            {
+                "name": "Channel IT",
+                "country_ids": [(4, cls.italy.id)],
+                "state_ids": [],
+                "state": "open",
+            }
+        )
+        cls.channel_uk = cls.env["stock.release.channel"].create(
+            {
+                # We create this to make sure it is *excluded* when searching channels
+                # for pickings linked to ``partner_it`` and ``partner_it_an``
+                "name": "Channel UK",
+                "country_ids": [(4, cls.env.ref("base.uk").id)],
+                "state_ids": [],
+                "state": "open",
+            }
+        )
+        cls.channel_it_an = cls.env["stock.release.channel"].create(
+            {
+                "name": "Channel IT - Ancona",
+                "country_ids": [(4, cls.italy.id)],
+                "state_ids": [(4, cls.ancona.id)],
+                "state": "open",
+            }
+        )
+        cls.channel_it_rm = cls.env["stock.release.channel"].create(
+            {
+                # We create this to make sure it is *excluded* when searching channels
+                # for pickings linked to ``partner_it_an``
+                "name": "Channel IT - Roma",
+                "country_ids": [(4, cls.italy.id)],
+                "state_ids": [(4, cls.roma.id)],
+                "state": "open",
+            }
+        )
+        cls.channel_it_an_rm = cls.env["stock.release.channel"].create(
+            {
+                # We create this to make sure it is *included* when searching channels
+                # for pickings linked to ``partner_it_an``
+                "name": "Channel IT - Ancona+Roma",
+                "country_ids": [(4, cls.italy.id)],
+                "state_ids": [(4, cls.ancona.id), (4, cls.roma.id)],
+                "state": "open",
+            }
+        )
+        cls.all_channels = (
+            cls.channel
+            + cls.channel_it
+            + cls.channel_it_an
+            + cls.channel_it_rm
+            + cls.channel_it_an_rm
+            + cls.channel_uk
+        )
+
+    def test_00_picking_find_release_channel_match_by_address_no_country_no_state(self):
```

Comment:
I don't think is necessary to number tests... but yeah

Reply:
No it isn't. Not with TransationCase.